crates/project_model/src/workspace.rs

```diff
@@ -227,6 +227,25 @@ impl ProjectWorkspace {
                             .and_then(|it| it.out_dir.clone()),
                     );
 
+                    let extra_targets: Vec<AbsPathBuf> = cargo[pkg]
+                        .targets
+                        .iter()
+                        .filter(|&&tgt| {
+                            cargo[tgt].kind == TargetKind::Lib
+                                && cargo[tgt]
+                                    .root
+                                    .parent()
+                                    .unwrap()
+                                    .parent()
+                                    .unwrap()
+                                    .to_path_buf()
+                                    .normalize()
+                                    != pkg_root
+                        })
+                        .map(|&tgt| cargo[tgt].root.parent().unwrap().to_path_buf())
+                        .collect();
+                    include.extend(extra_targets);
```

Comment:
```suggestion
                    // In case target's path is manually set in Cargo.toml to be
                    // outside the package root, at its parent as an extra include.
                    // An example of this situation would look like this:
                    //
                    // ```toml
                    // [lib]
                    // path = "../../src/lib.rs"
                    // ```
                    let extra_targets = cargo[pkg]
                        .targets
                        .iter()
                        .filter(|&&tgt| cargo[tgt].kind == TargetKind::Lib)
                        .filter_map(|&tgt| cargo[tgt].root.parent())
                        .filter(|path| !path.starts_with(pkg_root));
                    include.extend(extra_targets);
```

Rather than manually checking equality, let's check if a target's path is within the package. Would this work?

Reply:
Almost worked, much cleaner. thanks.